redis/client.go

```diff
@@ -216,7 +216,7 @@ func (c *Client) XReadUntilResult(ctx context.Context, a *redis.XReadArgs) ([]re
 		cmd := c.XRead(ctx, a)
 		streams, err := cmd.Result()
 		if err != nil {
-			if errors.Is(err, redis.Nil) {
+			if (errors.Is(err, redis.Nil) || retry.Retryable(err)) && ctx.Err() == nil {
```

Comment:
Coming back to this diff after just one week, I have totally forgotten why there is a `ctx.Err() == nil`. Could you please add a short comment with the reasoning from <https://github.com/Icinga/icinga-go-library/pull/23#discussion_r1705689655> and <https://github.com/Icinga/icinga-go-library/pull/23#discussion_r1707154827>. Otherwise, LGTM. Thanks.

Reply:
IMAO it's not limited to those comments. One could also pass a context and cancel it, regardless the current error. Then, it makes no sense to re-try.

But yes, `ctx.Err()==nil` is not our idiomatic way of checking whether ctx hasn't been e.g. cancelled, yet. What about instead code which doesn't need such commenting in the first place, i.e. select on ctx.Done() as usual?

Reply:
I can't follow why the following (i.e. the current version of the PR) wouldn't need an explaining comment when checking `ctx.Err() == nil` does:

https://github.com/Icinga/icinga-go-library/blob/2efd1f7f182912edaf40928a68a58b7d07a0d224/redis/client.go#L220-L224

It's pretty obvious to me that both does the same, the `select` is just more bloated.

> But yes, `ctx.Err()==nil` is not our idiomatic way of checking whether ctx hasn't been e.g. cancelled, yet. What about instead code which doesn't need such commenting in the first place, i.e. select on ctx.Done() as usual?

Is it idiomatic? It's simply necessary if you want to use a context to unblock waiting send/receive on a channel.

Reply:
@Al2Klimov can you please revert this to the previous version and just add a simple comment as to why it is necessary to check for `ctx.Error()`. Introducing a `select` case just to check if a ctx has an error instead of just adding a simple comment is ridiculous and keeps the PR open for nothing.